devito/arch/archinfo.py

```diff
@@ -302,6 +303,38 @@ def homogenise_gpus(gpu_infos):
     return None
 
 
+@memoized_func
+def detect_cc():
```

Comment:
this is an NVidia-only function, so I'd name it differently to denote that. For example, `get_nvidia_cc` ?